libdnf5/conf/option_enum.cpp

```diff
@@ -26,165 +26,86 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 namespace libdnf5 {
 
-template <typename T>
-bool from_string(T & out, const std::string & in, std::ios_base & (*manipulator)(std::ios_base &)) {
-    std::istringstream iss(in);
-    return !(iss >> manipulator >> out).fail();
-}
-
-template <typename T>
-OptionEnum<T>::OptionEnum(ValueType default_value, const std::vector<ValueType> & enum_vals)
+class OptionEnum::Impl {
+public:
+    Impl(std::string default_value, const std::vector<std::string> & enum_vals)
```

Comment:
Replace `const std::vector<std::string> & enum_vals` -> `std::vector<std::string> && enum_vals`
`const` cannot be moved out.